src/server/bitops_family.cc

```diff
@@ -577,12 +593,488 @@ void BitCount(CmdArgList args, ConnectionContext* cntx) {
+    CommandApplyVisitor visitor(std::move(value));
+    for (auto& command : commands) {
+      auto res = std::visit(visitor, command);
+      if (res) {
+        results.push_back(*res);
+      }
+    }
+
+    if (visitor.ShouldCommit()) {
+      access_.Find(shard_);
+      access_.Commit(visitor.Value());
+    }
+
+    return results;
+  }
+
+ private:
+  ElementAccess access_;
+  EngineShard* shard_;
+};
+
+nonstd::expected<CommonAttributes, std::string> ParseCommonAttr(CmdArgParser& parser) {
+  CommonAttributes parsed;
+  using nonstd::make_unexpected;
+  if (!parser.HasAtLeast(2)) {
+    return make_unexpected(kSyntaxErr);
+  }
+
+  auto encoding = parser.ToUpper().Next();
+  if (absl::StartsWith(encoding, "U")) {
+    parsed.type = EncodingType::UINT;
+  } else if (absl::StartsWith(encoding, "I")) {
+    parsed.type = EncodingType::INT;
+  } else {
+    return make_unexpected(kSyntaxErr);
+  }
+
+  std::string_view bits = encoding;
+  bits = bits.substr(1);
+
+  if (!absl::SimpleAtoi(bits, &parsed.encoding_bit_size)) {
+    return make_unexpected(kSyntaxErr);
+  }
+
+  if (parsed.encoding_bit_size <= 0 || parsed.encoding_bit_size > 64) {
+    return make_unexpected(
+        "invalid bitfield type. use something like i16 u8. note that u64 is not supported but i64 "
+        "is.");
+  }
+
+  if (parsed.encoding_bit_size == 64 && parsed.type == EncodingType::UINT) {
+    return make_unexpected(
+        "invalid bitfield type. use something like i16 u8. note that u64 is not supported but i64 "
+        "is.");
+  }
+
+  auto offset_proxy = parser.Next();
+  std::string_view offset_str = offset_proxy;
+  bool is_proxy = false;
+  if (absl::StartsWith(offset_str, "#")) {
+    offset_str = offset_str.substr(1);
+    is_proxy = true;
+  }
+  if (!absl::SimpleAtoi(offset_str, &parsed.offset)) {
+    return make_unexpected(kSyntaxErr);
+  }
+  if (is_proxy) {
+    parsed.offset = parsed.offset * parsed.encoding_bit_size;
+  }
+  return parsed;
+}
+
+nonstd::expected<int64_t, std::string> ParseValue(CmdArgParser& parser) {
+  using nonstd::make_unexpected;
+  if (!parser.HasAtLeast(1)) {
+    return make_unexpected(kSyntaxErr);
+  }
+
+  auto value = parser.ToUpper().Next();
+  int64_t res{0};
+  if (!absl::SimpleAtoi(value, &res)) {
+    return make_unexpected(kSyntaxErr);
+  }
+  return res;
+}
+
+nonstd::expected<CommandList, std::string> ParseToCommandList(CmdArgList args) {
+  CommandList result;
+
+  using nonstd::make_unexpected;
+
+  CmdArgParser parser(args);
+  while (parser.HasNext()) {
+    if (!parser.HasAtLeast(2)) {
+      return make_unexpected(kSyntaxErr);
+    }
+
+    auto op = parser.ToUpper().Next();
+
+    using namespace std::string_view_literals;
+    if (op == "OVERFLOW"sv) {
+      auto policy = parser.ToUpper().Next();
+      if (auto res = Overflow::PolicyFromStr(policy); res) {
+        result.push_back(Overflow{*res});
+        continue;
+      }
+      return make_unexpected(kSyntaxErr);
+    }
+
```

Comment:
1. We have `parser.Check()` to simplify tag checking
2. We have `parser.Switch()` so all of the ` Overflow::PolicyFromStr` is not needed

Reply:
+1 for `parser.check()`. However, the reason I did not user `Switch` was a) I find it a little bit more verbose b) I did not want to check if the parser failed (I find the optional a bit more expressive in this case).

Reply:
I'd suggest then to check at the end that the parser error is null (it fails silently so to check your wrapper)